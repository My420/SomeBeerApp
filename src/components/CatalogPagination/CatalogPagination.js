import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './CatalogPagination.module.scss';
import ListGroup from '../ListGroup/ListGroup';
import ListGroupItem from '../ListGroupItem/ListGroupItem';
import createOptions from '../../utils/createOptions';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  EBC_LESS_PROP,
  EBC_MORE_PROP,
  PAGE,
  BEER_NAME_PROP
} from '../../utils/constants';

const CatalogPagination = ({ options }) => {
  const page = options[PAGE];
  const prevPage = `${+page - 1}`;
  const nextPage = `${+page + 1}`;
  const isPrevDisabled = prevPage < 1;
  const prevLink = createOptions({ ...options, [PAGE]: prevPage });
  const nextLink = createOptions({ ...options, [PAGE]: nextPage });
  return (
    <ListGroup className={styles.list}>
      <div className={styles.wrapper}>
        <ListGroupItem className={styles.item}>
          {!isPrevDisabled ? (
            <Link to={prevLink} className={styles.prev} disabled={prevPage < 1}>
              prev
            </Link>
          ) : (
            <p className={styles.prevDisabled}>prev</p>
          )}
        </ListGroupItem>
        <ListGroupItem className={styles.item}>
          <Link to={nextLink} className={styles.next}>
            next
          </Link>
        </ListGroupItem>
      </div>
    </ListGroup>
  );
};

CatalogPagination.propTypes = {
  options: PropTypes.shape({
    [PAGE]: PropTypes.string.isRequired,
    [BEER_NAME_PROP]: PropTypes.string.isRequired,
    [ABV_LESS_PROP]: PropTypes.string.isRequired,
    [ABV_MORE_PROP]: PropTypes.string.isRequired,
    [IBU_LESS_PROP]: PropTypes.string.isRequired,
    [IBU_MORE_PROP]: PropTypes.string.isRequired,
    [EBC_LESS_PROP]: PropTypes.string.isRequired,
    [EBC_MORE_PROP]: PropTypes.string.isRequired
  }).isRequired
};

export default CatalogPagination;
