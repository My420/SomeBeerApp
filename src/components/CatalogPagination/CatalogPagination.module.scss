@import '../../sass/variables.scss';

.list {
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  height: 70px;

  padding-bottom: 20px;

  list-style: none;
}

.wrapper {
  display: flex;
}

.prev,
.next {
  @include link-button;
}

.next {
  margin-left: 40px;
}

.prevDesabled {
  display: flex;
  justify-content: center;
  align-items: center;

  padding: 3px 10px;

  box-sizing: border-box;

  border: 2px solid $menu-color;
  border-radius: 5px;
  background-color: transparent;

  font-size: 18px;
  line-height: 18px;
  color: $menu-color;

  text-decoration: none;
  cursor: default;
  opacity: 0.4;
}

@media (max-width: $Tablet7-max-width) {
  .list {
    height: 30px;

    padding-bottom: 30px;
  }
}
