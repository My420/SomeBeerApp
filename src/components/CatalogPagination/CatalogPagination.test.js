import React from 'react';
import { shallow } from 'enzyme';
import CatalogPagination from './CatalogPagination';
import createOptions from '../../utils/createOptions';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  EBC_LESS_PROP,
  EBC_MORE_PROP,
  PAGE,
  BEER_NAME_PROP
} from '../../utils/constants';

describe('test <CatalogPagination /> component', () => {
  const options = {
    [ABV_LESS_PROP]: '1',
    [ABV_MORE_PROP]: '2',
    [IBU_LESS_PROP]: '3',
    [IBU_MORE_PROP]: '4',
    [EBC_LESS_PROP]: '5',
    [EBC_MORE_PROP]: '6',
    [PAGE]: '1',
    [BEER_NAME_PROP]: 'test'
  };

  const wrapper = shallow(<CatalogPagination options={options} />);
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <p> tag as prev button if prev page < 1', () => {
    expect(wrapper.find('p.prevDisabled')).toHaveLength(1);
    expect(wrapper.find('p.prevDisabled').text()).toBe('prev');
  });

  test('should render  prev button if prev page > 0', () => {
    wrapper.setProps({ options: { ...options, [PAGE]: 3 } });
    expect(wrapper.find('p.prevDisabled')).toHaveLength(0);
    expect(wrapper.find('Link.prev')).toHaveLength(1);
    expect(wrapper.find('Link.prev').prop('children')).toBe('prev');
  });

  test('should render  next button', () => {
    expect(wrapper.find('Link.next')).toHaveLength(1);
    expect(wrapper.find('Link.next').prop('children')).toBe('next');
  });

  test("buttons should have correct prop 'to'", () => {
    const newOptions = { ...options, [PAGE]: '77' };
    wrapper.setProps({ options: newOptions });
    const prev = createOptions({ ...newOptions, [PAGE]: '76' });
    const next = createOptions({ ...newOptions, [PAGE]: '78' });
    expect(wrapper.find('Link.next').prop('to')).toBe(next);
    expect(wrapper.find('Link.prev').prop('to')).toBe(prev);
  });
});
