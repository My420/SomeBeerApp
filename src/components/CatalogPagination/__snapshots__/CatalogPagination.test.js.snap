// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <CatalogPagination /> component should match snapshot 1`] = `
<ListGroup
  className="list"
  tag="ul"
>
  <div
    className="wrapper"
  >
    <ListGroupItem
      className="item"
    >
      <p
        className="prevDisabled"
      >
        prev
      </p>
    </ListGroupItem>
    <ListGroupItem
      className="item"
    >
      <Link
        className="next"
        to="?page=2&per_page=24&beer_name=test&abv_lt=1&abv_gt=2&ibu_lt=3&ibu_gt=4&ebc_lt=5&ebc_gt=6"
      >
        next
      </Link>
    </ListGroupItem>
  </div>
</ListGroup>
`;
